src/session-pool.ts

```diff
@@ -1036,6 +1151,71 @@ export class SessionPool extends EventEmitter implements SessionPoolInterface {
     session.txn = transaction;
   }
 
+  /**
+   * recycles sessions with long-running transactions
+   *
+   * @private
+   *
+   * @param {number} longRunningTransactionTimeout Timeout for recycling sessions.
+   */
+  async _deleteLongRunningTransactions(
+    longRunningTransactionTimeout: number = 1000 * 60 * 60
+  ): Promise<void> {
+    // Check lock, only one cleanup should take place at a time
+    if (this._ongoingTransactionDeletion) return;
```

Comment:
Add a test to verify that multiple background tasks are not getting triggered. Something like below
```
session1 = getsession()
session2= getsession()
await promise.all(session1, session2)
```